esp/esdllib/esdl_script.cpp

```diff
@@ -226,6 +338,152 @@ class CEsdlTransformOperationSetValue : public CEsdlTransformOperationBase
+            else
+                path.set(m_target);
+
+            StringBuffer name;
+            if (m_xpath_new_name)
+                xpathContext->evaluateAsString(m_xpath_new_name, name);
+            else
+                name.set(m_new_name);
+
+            //make sure we only remove one.  we can add a multi-node support in the future (when we are libxml2 based?)
+            //don't want users to depend on quirky behavior
+            path.append("[1]"); //don't need to check as multiple [] are ok
+            content->renameProp(path, name);
+        }
+        catch (IException* e)
+        {
+            int code = e->errorCode();
+            StringBuffer msg;
+            e->errorMessage(msg);
+            e->Release();
+            esdlOperationError(code, m_tagname, msg, m_traceName, !m_ignoreCodingErrors);
+        }
+        catch (...)
+        {
+            esdlOperationError(ESDL_SCRIPT_Error, m_tagname, "unknown exception processing", m_traceName, !m_ignoreCodingErrors);
+        }
+        return false;
+    }
+};
+
+class CEsdlTransformOperationRemoveNode : public CEsdlTransformOperationBase
+{
+protected:
+    StringAttr m_target;
+    Owned<ICompiledXpath> m_xpath_target;
+
+public:
+    CEsdlTransformOperationRemoveNode(IPropertyTree *tree, const StringBuffer &prefix) : CEsdlTransformOperationBase(tree, prefix)
+    {
+        const char *target = tree->queryProp("@target");
+        const char *xpath_target = tree->queryProp("@xpath_target");
+        if (isEmptyString(target) && isEmptyString(xpath_target))
+            esdlOperationError(ESDL_SCRIPT_MissingOperationAttr, m_tagname.str(), "without target", m_traceName.str(), !m_ignoreCodingErrors);
+        if (target && isWildString(target))
+            esdlOperationError(ESDL_SCRIPT_MissingOperationAttr, m_tagname.str(), "wildcard in target not yet supported", m_traceName.str(), !m_ignoreCodingErrors);
+
+        if (!isEmptyString(xpath_target))
+            m_xpath_target.setown(compileXpath(xpath_target));
+        else if (!isEmptyString(target))
+            m_target.set(target);
+    }
+
+    virtual void toDBGLog() override
+    {
+#if defined(_DEBUG)
+        const char *target = (m_xpath_target) ? m_xpath_target->getXpath() : m_target.str();
+        DBGLOG(">%s> %s(%s)", m_traceName.str(), m_tagname.str(), target);
+#endif
+    }
+
+    virtual ~CEsdlTransformOperationRemoveNode(){}
+
+    virtual bool process(IEsdlScriptContext * context, IPropertyTree *content, IXpathContext * xpathContext) override
+    {
+        if ((!m_xpath_target && m_target.isEmpty()))
+            return false; //only here if "optional" backward compatible support for now (optional syntax errors aren't actually helpful
+        try
+        {
+            StringBuffer path;
+            if (m_xpath_target)
+                xpathContext->evaluateAsString(m_xpath_target, path);
+            else
+                path.set(m_target);
+
+            if (isWildString(path))
```

Comment:
It seems like we should just evaluate the path and delete a single node, if that is what getElements gives us. Unless this required because getElements can't evaluate the wildcard.

Reply:
We are not calling getElements. If I remember correclty removeProp did not like wildcards.

Reply:
I know it isn't using getElements. This was my way of suggesting it should use getElements. I just think taking '*' away, which is useful for more than looking for multiples, is undesirable.